### PR TITLE
refactor(core): Simplify Redis client types (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
- Remove unused client types: `client`, `bclient`, `consumer`, `producer`, `list-sender`, `list-received`
- Suffix n8n client types to differentiate them from Bull client types
- Document remaining client types

Context: https://n8nio.slack.com/archives/C069HS026UF/p1723618011118889

### CODE REVIEW

File: packages/cli/src/services/redis/redis.types.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export type RedisClient = n8nRedisClient | BullRedisClient;
```
Comment (tomi) [inline packages/cli/src/services/redis/redis.types.ts:1]: I believe this is the redis client _type_ (or name?), not the client itself. Maybe we should rename this to `RedisClientType` and `N8nRedisClientType` and `BullRedisClientType` (or `*Name`?)

File: packages/cli/src/services/redis/redis.types.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * Redis client used by n8n.
+ *
+ * - `subscriber(n8n)` to listen for messages from scaling mode communication channels
+ * - `publisher(n8n)` to send messages into scaling mode communication channels
+ * - `cache(n8n)` for caching operations (variables, resource ownership, etc.)
+ */
+type n8nRedisClient = 'subscriber(n8n)' | 'publisher(n8n)' | 'cache(n8n)';
```
Comment (tomi) [inline packages/cli/src/services/redis/redis.types.ts:10]: TS types should start with a capital letter, even if it's for a name that is normally written with a lowercase letter

File: packages/cli/src/services/redis/redis.types.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * Redis client used internally by Bull. Suffixed with `(bull)` at `ScalingService.setupQueue`.
+ *
+ * - `subscriber(bull)` for event listening
+ * - `client(bull)` for general queue operations
+ * - `bclient(bull)` for blocking operations when processing jobs
+ */
```
Comment (tomi) [inline packages/cli/src/services/redis/redis.types.ts:18]: Lovely documentation ❤️

**Review by tomi — COMMENTED** [review]
Nice job 👏 Couple suggestions regarding naming things 😇

**Review by tomi — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.56.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.56.0)

**Pull request merged**